csrc/adam/cpu_adam.cpp

```diff
@@ -102,11 +103,16 @@ void Adam_Optimizer::Step(float* _params,
             SIMD_STORE(_exp_avg + i, momentum_4.data);
             SIMD_STORE(_exp_avg_sq + i, variance_4.data);
         }
+#if defined(__ENABLE_CUDA__)

```

Comment:
I believe all the cuda and cublas header files should be guarded as well.  We would probably need to test the build in a CPU-only environment.

Reply:
I agree, I will add more guards for them too.